CMakeLists.txt

```diff
@@ -303,8 +304,14 @@ ELSE()
       SET(PA_PUBLIC_INCLUDES ${PA_PUBLIC_INCLUDES} include/pa_linux_alsa.h)
       SET(PA_SOURCES ${PA_SOURCES} ${PA_ALSA_SOURCES})
       SET(PA_PRIVATE_COMPILE_DEFINITIONS ${PA_PRIVATE_COMPILE_DEFINITIONS} PA_USE_ALSA)
-      SET(PA_LIBRARY_DEPENDENCIES ${PA_LIBRARY_DEPENDENCIES} ${ALSA_LIBRARIES})
-      SET(PA_PKGCONFIG_LDFLAGS "${PA_PKGCONFIG_LDFLAGS} -lasound")
+      IF(PA_ALSA_DYNAMIC)
+        SET(PA_PRIVATE_COMPILE_DEFINITIONS ${PA_PRIVATE_COMPILE_DEFINITIONS} PA_ALSA_DYNAMIC)
+        SET(PA_LIBRARY_DEPENDENCIES ${PA_LIBRARY_DEPENDENCIES} ${CMAKE_DL_LIBS})
+        SET(PA_PKGCONFIG_LDFLAGS "${PA_PKGCONFIG_LDFLAGS} -l${CMAKE_DL_LIBS}")
```

Comment:
Why did you add these 2 lines? I believe they are not needed for the proposed functionality, only `PA_ALSA_DYNAMIC` needs to be defined without any other dependencies added. It has to be corrected.

Reply:
no, on my machine it fails to link in that case as it does not find dlopen / dlsym.

Reply:
Ok, understood, I overlooked that dl library is not used by PA by default. Then agree that these 2 lines are needed.